app.ts

```diff
@@ -0,0 +1,323 @@
+	return fetchRecords(fromRecord, toRecord)
+		.then((records) => {
+			let foundIndex = -1;
+			let recordIndex = 0;
+			for (const record of records) {
+				const idValue = parseInt(record[0]);
+				if (idValue === searchValue) {
+					foundIndex = fromRecord + recordIndex;
+					break;
+				}
+				recordIndex++;
+			}
+			if (foundIndex !== -1) {
+				state.searchedIndex = foundIndex;
+				currentPage = searchPage;
+				state.currentFirstRecordIndex = (searchPage - 1) * state.recordsPerPage;
+				return displayData(state.currentFirstRecordIndex, state.recordsPerPage);
+			} else {
+				state.searchedIndex = null;
+			}
+			return Promise.resolve();
+		})
+		.then(() => {
+			$("#loader").hide();
+			$("#tableWrapper").show();
+			$("#prevPageButton").show();
+			if (searchPage === totalPages) {
+				$("#nextPageButton").hide();
+			} else {
+				$("#nextPageButton").show();
+			}
+		}).catch((error) => {
+			throw error;
+		})
+}
+
+// Update the screen layout and display records based on the current screen/window size
+async function updateScreen() {
+	const newScreenHeight = window.innerHeight;
+	totalRecords();
+	state.recordsPerPage = screenCalculations(newScreenHeight);
+	$("#loader").show();
+	$("#tableWrapper").hide();
+	if (state.searchedIndex !== null) {
+		const searchPage = Math.ceil((state.searchedIndex + 1) / state.recordsPerPage);
+		const searchPageIndex = state.searchedIndex % state.recordsPerPage;
+		state.searchedIndex = (searchPage - 1) * state.recordsPerPage + searchPageIndex;
+		displayData(Math.max(state.searchedIndex - state.recordsPerPage + 1, 0), state.recordsPerPage);
+	} else {
+		const previousFirstRecordIndex = state.currentFirstRecordIndex;
+		currentPage = Math.ceil((state.currentFirstRecordIndex + 1) / state.recordsPerPage);
+		displayData(state.currentFirstRecordIndex, state.recordsPerPage);
+	}
+	$("#loader").hide();
+	$("#tableWrapper").show();
+	if (currentPage === totalPages) {
+		$("#nextPageButton").hide();
+	} else {
+		$("#nextPageButton").show();
+	}
+	if (state.currentFirstRecordIndex <= 1) {
+		$("#prevPageButton").hide();
+	} else {
+		$("#prevPageButton").show();
+	}
+}
+
+window.onload = async function () {
+	$("#loader").hide();
+	updateScreen();
+	fetchColumns();
+	inputHandling();
+	$(window).on("resize", debounce(async () => {
+		$("#tableWrapper").hide();
+		$("#loader").show();
+		updateScreen()
+			.then(async () => {
+				const totalRecCount = await totalRecords();
+				if (currentPage * state.recordsPerPage > totalRecCount - 1) {
+					const lastPage = Math.ceil(totalRecCount / state.recordsPerPage);
+					currentPage = lastPage;
+					state.currentFirstRecordIndex = (lastPage - 1) * state.recordsPerPage;
+					return displayData(state.currentFirstRecordIndex, state.recordsPerPage);
+				}
+			})
+			.then(() => {
+				$("#loader").hide();
+				$("#tableWrapper").show();
+			}).catch((error) => {
+				throw (error);
+			});
+	}, 150))
+
+	$("#searchInput").on("keydown", function (e) {
```

Comment:
```suggestion
	$("#searchInput").on("keydown", (e) => {
```
We prefer TypeScript arrow functions.